xapian-glib/xapian-database.cc

```diff
@@ -808,3 +810,25 @@ xapian_database_get_flags (XapianDatabase *self)
 
   return db_flags;
 }
+
+/**
+ * xapian_database_allterms:
+ * @self: A #XapianDatabase
+ * @prefix: (nullable): Prefix to iterate over
+ *
+ * Start iterating over all terms in the database, optionally restricted to
+ * terms with the specified prefix.
+ *
+ * Returns: (transfer full): a #XapianTermIterator instance
+ *
+ * Since: 1.4
+ */
+XapianTermIterator *
+xapian_database_allterms (XapianDatabase *self,
```

Comment:
Well, it's obviously possible, though I'd suggest `xapian_database_enumerate_all_terms()` so (a) it's clearer what it's doing, and (b) it's more obvious what it's equivalent to for people trying to follow example code in a different language (especially useful as there aren't any tutorials, etc specific to xapian-glib AFAICS).

This also means that the already merged code in e-c-f would need changing.